### ISSUE TITLE
Onboarding review

### ISSUE THREAD

**Issue by mjourdan** [lutris/lutris#2880]
Recently, a relative of mine wanted to play some games purchased on GoG. I remembered I read about an app to manage game collections on Linux, so I encouraged her to give it a try.

So, you'll find here a report of our first experience with Lutris, what we did and how well things went. 

### First step, getting Lutris

First thing first, if I did remember reading about Lutris on some website, I couldn't remember the actual name of the app. So, I browsed a bit Flathub, but couldn't find what I was looking for. Then, I had to search a bit for "linux gaming" in DuckDuckGo, and finally had interesting results.

We visited Lutris website, went to the "Download" section, and got kind of lost on this page. On the one hand we see how system packaging fragmentation is harmful. On the other hand, it is quite encouraging to see Lutris is packaged for many distributions.

So, let's open a terminal and type in the command to install Lutris on Fedora!

### Second step, launching

Alright, Lutris is installed, let's run it from the shell. We typed `lutris`, and expected a window to open. Instead, we got a pop-up asking for donations. Duh. Maybe later, thank you. We click on `Validate` and the actual app opens.

### Third step, what to do next?

Ok now, we are facing a window. 3 columns, almost empty, and no clue on what to do next. The 1st column seems to list "runners" (is that a gaming thing or something?), the 2nd column is blank, and the 3rd column says "support Lutris!". Again!? So far, Lutris hasn't helped to make running games easier, so, this feels quite pressuring. Oh, wait! Texts in the 3rd column happens to look like buttons on hover! 

So, we were here to play our games purchases on GoG, remember? So, maybe we shall fill in our Gog account? Let's click this account button in the top bar! "Login" or "register"… "Login"! A dialog opens "Connect to lutris.net". What the hell is this for? We go back to the Lustris website, looking for clues about this mysterious account… No idea of what this is used for. Is this required or something?

Ok, Let's see what is behind the "+" button in the top bar. "Add games"… Nope, too many settings. "Import games" should do better. Looking attentively, we recognize the GoG logo in the left bar! Let's "connect to our account"! A pop-up window opens, let's enter our credentials! Our games appear! Let's select the whole list, and import everything! 

### Fourth step, installing games

In the main window, pick up a game and click "install". We expected the installation to happen immediately, but instead had to pick up GoG amongst a couple of items (including Steam). What is this list for? Is it to pick up the Game from different websites? Why would we want to download from Steam the game we purchased on GoG? Is it even possible?

### Fifth step, playing!

Can we just select the game we just installed, and click play? Phew! I love it when a plan comes together! 

### Recapitulation

To sum up, we identified key areas to improve Lutris user onboarding. These are the following:
- installation process (allow people to fetch Lutris from their preferred app repository, as non-techie people don't even know what their distribution is)
- donation messages (don't be too pushy, do not ask for donations before people couldn't not even do anything with the app)
- providing guidance (manage empty states, give clues on what to do next)
- avoiding unnecessary complexity (ie, import and install should happen from the same source)
 
Hope this give highlights on what can go wrong for newcomers, and help to make first steps with Lutris more pleasant.

**Comment by xordspar0** [comment]
> donation messages (don't be too pushy, do not ask for donations before people couldn't not even do anything with the app)

I agree, it could be helpful to delay this message until the user has successfully used Lutris a couple times.

> donation messages (don't be too pushy, do not ask for donations before people couldn't not even do anything with the app)

Yes, and maybe clarify the difference between "add" and "import".

> avoiding unnecessary complexity (ie, import and install should happen from the same source)

There is definitely room for improvement. I don't have any specific suggestions, but I will note that GOG integration has only been in Lutris for a few releases.

**Comment by mjourdan** [comment]
To make the first run as smooth as possible, I made some wireframes I would be happy to discuss:

![first run](https://raw.githubusercontent.com/mjourdan/lutris-design/master/lutris.png)

Even though there is a fair amount of changes, some parts are not covered:
- the Lutris account (as I have no idea what this is for)
- the call to donation (because a first run is probably too early for this)
- the process for adding a single game
- other matters not strictly related to the first run (the preferences window, the game details panel…) 

Feel free to comment!

**Comment by mjourdan** [comment]
For your information, wireframes have been updated to cover install and play steps.

**Comment by Chugajstyr** [comment]
> but instead had to pick up GoG amongst a couple of items (including Steam). What is this list for? Is it to pick up the Game from different websites? Why would we want to download from Steam the game we purchased on GoG? Is it even possible?

Have you read [Lutris About](https://lutris.net/about/) page? I mean, I started to use Lutris because of GOG integration too, but Lutris positioned as Linux gaming client not only for GOG games. I personally have a few games purchased both on GOG as well as on Steam. So, Lutris installer didn't recognize (yet) that game you installing is from GOG, not Steam, or other platform - but it allows you to make the choice.  :)

> the Lutris account (as I have no idea what this is for)

Currently it's store your imported games library.

P.S. Btw, I like your wireframes. Before Lutris I used [PlayOnLinux](https://www.playonlinux.com/en/), I loved their UI, but that client doesn't have import from different platforms or runners features. And their project seems stagnated now.

**Comment by strycore** [comment]
Thanks for the in depth review and the mockups, that's appreciated a lot!
A lot of the points you raised should be addressed in the future releases.

Btw, if any one has any idea on how to remove the right sidebar while keeping the functionality, I'm really open to suggestions.

**Comment by mjourdan** [comment]
Thanks people for your feedback! Answers to a few points below.

> Yes, and maybe clarify the difference between "add" and "import".

Yep. To do so, the mockups allow to add either full libraries or single games. Imports would happen automatically when a library is added (and on a few different occasions also, like on startup, but you get the idea).
 
> There is definitely room for improvement. I don't have any specific suggestions, but I will note that GOG integration has only been in Lutris for a few releases.

Sure. I have no idea how advanced/behind GOG integration is in comparison with others, but it certainly help Lutris to reach a wider audience a great deal. Even though I took GOG as an example, I hope Lutris can deliver a seamless experience for a common feature set (adding libraries, importing games, installing…), no matter how the technical implementation differs.

> Have you read [Lutris About](https://lutris.net/about/) page? I mean, I started to use Lutris because of GOG integration too, but Lutris positioned as Linux gaming client not only for GOG games. I personally have a few games purchased both on GOG as well as on Steam. So, Lutris installer didn't recognize (yet) that game you installing is from GOG, not Steam, or other platform - but it allows you to make the choice. :)

I did! It is indeed a good thing to allow to choose from which platform the binaries should be fetch, but it is only needed when the game has been imported from more than one library. Otherwise it is irrelevant.

> Currently it's store your imported games library.

Ah, ok! I noticed on lutris.net a number of users was shown for each game, I figured it had something to do with the account. I would be curious to know more about the goals behind this, so maybe I could make some suggestion. Anyway I guess I'll open a separate issue for that.

> Thanks for the in depth review and the mockups, that's appreciated a lot!
> A lot of the points you raised should be addressed in the future releases.

That's good news! Do not hesitate if I can help further on this.

> Btw, if any one has any idea on how to remove the right sidebar while keeping the functionality, I'm really open to suggestions.

Which functionality? Are you talking about the support/help panel, or the game panel? If it is the former, all items but irc have just been moved to the burger menu. If it is the later, the sidebar would be replaced by the full-width view. But maybe it is the possibility to launch the game without leaving the "main view" you would like to keep? In any case, I would be happy to give a hand on this.

**Comment by strycore** [comment]
One of the main features from the sidebar I'd like to keep is the ability to see running games. 
That could be moved to a filter in the left sidebar so that's pretty easy to solve.
Another feature I'd like to have is the ability to have all game actions visible when a game is selected.
I don't really care about the support links and all of that. Those were mostly added so there wouldn't be a huge empty space when no game is selected.
The initial idea I had was to have a sidebar shown only when a game is selected and no sidebar otherwise, similar to what Origin does. I didn't manage to get that implemented with Gtk and it wouldn't have provided me with a space to put running games so I went with a permanent right sidebar.
Having a view for games taking the whole space causes navigation problems. If that shows up only for running games then a lot of features are hidden under a right click. If that view is displayed for each selected games then it involves a lot of back and forward when browsing through games. UPlay does that but since it's usually used for a much smaller game library, it's not that big of a problem.

**Comment by mjourdan** [comment]
Thanks for the detailed explanations. I wasn't aware the sidebar was meant for listing running games, because when a game is selected then launched, the sidebar got obfuscated by the game panel.

Below are some experimental mockups to adress that and the concerns you raised. Also, an idea for some in-app call to donation. Feel free to comment!
![lutris-experiments](https://raw.githubusercontent.com/mjourdan/lutris-design/master/lutris-experiments.png)

**Comment by kapcom01** [comment]
I really like these mockups, and I believe it is important to have a more intuitive GUI even for the non newcomers. I often find myself confused by the organization of the GUI.. for example I always forget where is the runners manager and which button to click to import a new game I bought on gog.. I believe, new games should be imported automatically under the respective Library as seen in the mockups.

>  Another feature I'd like to have is the ability to have all game actions visible when a game is selected.

Just a suggestion, I think it makes sense to move _Consoles & Platforms_ from the sidebar to the _Toggle View_ menu, as it is more like a filter rather than like a folder/store/repository. Then there would be plenty of free space to have all the game options of the selected game.

**Comment by strycore** [comment]
The good news is that some of the aspect of those mockups are planned to land in the next release.
I have integrated the "Favorites" features, meaning we now have a "Library" section at the top of the sidebar.

This will integrate the game store sections, meaning this will get rid of the "import games" dialog.

**Comment by battlegroundspng** [comment]
New hire induction is an important part of the employment lifecycle, especially early on. It’s a good idea to ask employees for feedback on how enabled they feel to complete their work and their feelings about the expected learning curve, You can also gain an understanding of how your new employee feels about the training plan they were assigned to get up to speed, their perceptions of their manager, and whether they had satisfactory opportunities to relate with other new and seasoned employees.

I'm confident in using the systems I need in my role. [Rating scale]
I have had good training on the processes applicable to my role. [Rating scale]
I have a good idea about what I still need to learn to do my job well. [Rating scale]
The information provided has been at the right level for me. [Rating scale]
[Battlegrounds Png](https://www.battlegroundspng.com)
